### PR TITLE
Add ClientError Exception

### LINKED ISSUE #2: Raise specific exceptions on 400
For example, pass `InvalidOption` into `chart` and you get a 400 with a JSON response. Raise something better than a generic `Faraday::ClientError: the server responded with status 400`.

### PR DESCRIPTION
**Pull request by rodolfobandeira**
close #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -146,6 +146,14 @@ See [#chart](https://iextrading.com/developer/docs/#chart) for detailed document
 
 If a symbol cannot be found an [IEX::Errors::SymbolNotFound](lib/iex/errors/symbol_not_found_error.rb) exception is raised.
 
+### BadRequest
+
+All 400 bad request responses from IEX result in a BadRequest exception
```
Comment (dblock) [inline README.md]: Nitpick: change `BadRequest` to `IEX::Errors::BadRequest` as above and link it to the .rb file.

File: lib/iex/resources/chart.rb [hunk 2]
```diff
@@ -21,6 +21,8 @@ def self.get(symbol, range = nil, options = {})
         end
       rescue Faraday::ResourceNotFound => e
         raise IEX::Errors::SymbolNotFoundError.new(symbol, e.response[:body])
+      rescue Faraday::ClientError => e
+        raise IEX::Errors::BadRequestError, e.response[:body]['error']
```
Comment (dblock) [inline lib/iex/resources/chart.rb]: There's a bug here. You're passing in the text of the error, but the exception itself takes a `response` object. This should be `e.response` and the exception should be doing `super e.response[:body]['error']`.

File: spec/iex/resources/chart_spec.rb [hunk 3]
```diff
@@ -72,4 +77,14 @@
...
     end
   end
+
+  context 'with bad request', vcr: { cassette_name: 'chart/badRequest' } do
+    subject do
+      IEX::Resources::Chart.get('MSFT', '1d', chart_interval: 10, invalid: 'option')
+    end
+
+    it 'fails with BadRequestError' do
+      expect { subject }.to raise_error IEX::Errors::BadRequestError
```
Comment (dblock) [inline spec/iex/resources/chart_spec.rb]: You should be checking that the exception caught has a `.response` object and that its error message is the extracted message from the body.

**Comment by rodolfobandeira** [comment]
Hey @dblock. Renamed to `ClientError`.

File: README.md [hunk 4]
```diff
@@ -146,6 +146,14 @@ See [#chart](https://iextrading.com/developer/docs/#chart) for detailed document
 
 If a symbol cannot be found an [IEX::Errors::SymbolNotFound](lib/iex/errors/symbol_not_found_error.rb) exception is raised.
 
+### ClientError
+
+All 400-600 bad request responses from IEX result in a [IEX::Errors::Client](lib/iex/errors/client_error.rb) exception.
```
Comment (dblock) [inline README.md]: Type: `ClientError`, not `Client`.

Comment (dblock) [inline README.md]: They are not "bad request responses", say "All errors that return HTTP codes 400-600 result in ..."

File: lib/iex/version.rb [hunk 5]
```diff
@@ -1,3 +1,3 @@
 module IEX
-  VERSION = '0.3.4'.freeze
+  VERSION = '0.3.5'.freeze
```
Comment (dblock) [inline lib/iex/version.rb]: Undo this.

File: spec/iex/resources/chart_spec.rb [hunk 6]
```diff
@@ -72,4 +77,14 @@
       expect { subject }.to raise_error IEX::Errors::SymbolNotFoundError, 'Symbol INVALID Not Found'
     end
   end
+
+  context 'with client error', vcr: { cassette_name: 'chart/badRequest' } do
```
Comment (dblock) [inline spec/iex/resources/chart_spec.rb]: Lets rename the `badRequest` file here to `bad_option` to match the name of the bad option.

**Comment by rodolfobandeira** [comment]
@dblock All good! Thank YOU!

**Comment by dblock** [comment]
Merged.

**Pull request merged**